src/App/Pages/Accounts/HomePageViewModel.cs

```diff
@@ -144,5 +162,56 @@ await _stateService.IsAuthenticatedAsync(userId))
                 await _platformUtilsService.ShowDialogAsync(AppResources.GenericErrorMessage, AppResources.AnErrorHasOccurred, AppResources.Ok);
             }
         }
+
+        public async Task ShowEnvironmentPickerAsync()
+        {
+            var options = _displayEuEnvironment
+                    ? new string[] { AppResources.US, AppResources.EU, AppResources.SelfHosted }
+                    : new string[] { AppResources.US, AppResources.SelfHosted };
+
+            await Device.InvokeOnMainThreadAsync(async () =>
+            {
+                var result = await Page.DisplayActionSheet(AppResources.DataRegion, AppResources.Cancel, null, options);
+
+                if (result is null || result == AppResources.Cancel)
+                {
+                    return;
+                }
+
+                if (result == AppResources.SelfHosted)
+                {
+                    StartEnvironmentAction?.Invoke();
+                }
+                else
+                {
+                    await _environmentService.SetUrlsAsync(result == AppResources.EU ? EnvironmentUrlData.DefaultEU : EnvironmentUrlData.DefaultUS);
+                    SelectedEnvironmentName = result;
+                }
+            });
+        }
+
+        public async Task UpdateEnvironment()
+        {
+            var environmentsSaved = await _stateService.GetPreAuthEnvironmentUrlsAsync();
+            if (environmentsSaved == null || environmentsSaved.IsEmpty())
```

Comment:
❓ Which would be the case the `EnvironmentUrlData` is not `null` but empty? Is that when the user filled the env data and then edited it and clear it out?

Reply:
yes, when the user saves the env page empty.